tests/sentry/integrations/gitlab/test_repository.py

```diff
@@ -203,3 +218,98 @@ def test_on_delete_repository_remove_webhook_missing_hook(self):
         repo = Repository.objects.get(pk=response.data['id'])
         self.provider.on_delete_repository(repo)
         assert len(responses.calls) == 1
+
+    @responses.activate
+    def test_compare_commits_start_and_end(self):
+        responses.add(
+            responses.GET,
+            'https://example.gitlab.com/api/v4/projects/%s/repository/compare?from=abc&to=xyz' % self.gitlab_id,
+            json=json.loads(COMPARE_RESPONSE)
+        )
+        responses.add(
+            responses.GET,
+            'https://example.gitlab.com/api/v4/projects/%s/repository/commits/12d65c8dd2b2676fa3ac47d955accc085a37a9c1/diff' % self.gitlab_id,
+            json=json.loads(COMMIT_DIFF_RESPONSE)
+        )
+        responses.add(
+            responses.GET,
+            'https://example.gitlab.com/api/v4/projects/%s/repository/commits/8b090c1b79a14f2bd9e8a738f717824ff53aebad/diff' % self.gitlab_id,
+            json=json.loads(COMMIT_DIFF_RESPONSE)
+        )
+        response = self.create_repository(self.default_repository_config,
+                                          self.integration.id)
+        repo = Repository.objects.get(pk=response.data['id'])
+        commits = self.provider.compare_commits(repo, 'abc', 'xyz')
+        assert 2 == len(commits)
+        for commit in commits:
+            assert_commit_shape(commit)
+
+    @responses.activate
+    def test_compare_commits_start_and_end_gitlab_failure(self):
```

Comment:
Accounting for the failure in test = 👍. Do we do this in other tests?

Reply:
Not consistently across integrations. Some of the integration repository tests patch the client method return value, so there is not even a simulated network call being made.